tests/RegistryTest.php

```diff
@@ -795,4 +795,55 @@ public function testCustomPathSeparatorsCanBeUsed()
         $this->assertEquals('test1', $a->get('Foo/Bar'));
         $this->assertEquals('test2', $a->get('Foo/Baz'));
     }
+
+    /**
+     * @testdox  The Registry operates correctly with empty string separator
+     *
+     * @covers   \Joomla\Registry\Registry
+     */
+    public function testEmptySeparator()
+    {
+        $a = new Registry(['foo' => 'bar']);
+        $a->separator = '';
+        $a->set('bar', 'beer');
+        $a->set('foo2.bar', 'beer');
+
+        $this->assertEquals('bar', $a->get('foo'));
+        $this->assertEquals('beer', $a->get('bar'));
+        $this->assertEquals('beer', $a->get('foo2.bar'));
+    }
+
+    /**
+     * @testdox  The Registry operates correctly with null separator
+     *
+     * @covers   \Joomla\Registry\Registry
+     */
+    public function testNullSeparator()
+    {
+        $a = new Registry(['foo' => 'bar']);
+        $a->separator = null;
+        $a->set('bar', 'beer');
+        $a->set('foo2.bar', 'beer');
```

Comment:
Use a different value here, e.g., 'wine'. Otherwise, you can't tell, if the beginning of the pass is just ignored.

Reply:
Makes sense, updated